v2/writer.go

```diff
@@ -0,0 +1,125 @@
+package car
+
+import (
+	"bytes"
+	"context"
+	"github.com/ipfs/go-cid"
+	format "github.com/ipfs/go-ipld-format"
+	car_v1 "github.com/ipld/go-car"
+	"github.com/willscott/carbs"
+	"io"
+)
+
+type (
+	// Padding represents the number of padding bytes.
+	Padding uint64
+	// Writer writes CAR v2 into a give io.Writer.
+	Writer struct {
+		Walk         car_v1.WalkFunc
+		IndexCodec   carbs.IndexCodec
+		NodeGetter   format.NodeGetter
+		CarV1Padding Padding
+		IndexPadding Padding
+
+		ctx   context.Context
+		roots []cid.Cid
+	}
+)
+
+// WriteTo writes this padding to the given writer as default value bytes.
+func (p Padding) WriteTo(w io.Writer) (n int64, err error) {
+	paddingBytes := make([]byte, p)
+	written, err := w.Write(paddingBytes)
+	n = int64(written)
+	return
+}
+
+// NewWriter instantiates a new CAR v2 writer.
+// The writer instantiated uses `carbs.IndexSorted` as the index codec,
+// and `car_v1.DefaultWalkFunc` as the default walk function.
+func NewWriter(ctx context.Context, ng format.NodeGetter, roots []cid.Cid) *Writer {
+	return &Writer{
+		Walk:       car_v1.DefaultWalkFunc,
+		IndexCodec: carbs.IndexSorted,
+		NodeGetter: ng,
+		ctx:        ctx,
+		roots:      roots,
+	}
+}
+
+// WriteTo writes the given root CIDs according to CAR v2 specification, traversing the DAG using the
+// Writer#Walk function.
+func (w *Writer) WriteTo(writer io.Writer) (n int64, err error) {
+	n, err = w.writePrefix(writer)
+	if err != nil {
+		return
+	}
+	// We read the entire car into memory because carbs#GenerateIndex takes a reader.
```

Comment:
use a period instead of #, since that's the Go syntax. I assume you used the godoc/pkgsite URL syntax here, but IMO most people expect just Go syntax in docs.